hyperspy/signal.py

```diff
@@ -76,6 +76,128 @@
 from hyperspy.drawing.utils import animate_legend
 from hyperspy.misc.hspy_warnings import VisibleDeprecationWarning
 from hyperspy.misc.slicing import SpecialSlicers, FancySlicing
+from hyperspy.misc.utils import slugify
+from datetime import datetime
+
+
+class ModelManager(object):
+
+    """Container for models
+    """
+
+    class ModelStub(object):
+
+        def __init__(self, mm, name):
+            self._name = name
+            self.restore = lambda: mm.restore(self._name)
+            self.remove = lambda: mm.remove(self._name)
+
+    def __init__(self, signal, dictionary=None):
+        self._signal = signal
+        self._models = DictionaryTreeBrowser()
+        self._add_dictionary(dictionary)
+
+    def _add_dictionary(self, dictionary=None):
+        if dictionary is not None:
+            for k, v in dictionary.iteritems():
+                if k.startswith('_') or k in ['restore', 'remove']:
+                    raise KeyError("Can't add dictionary with key '%s'" % k)
+                k = slugify(k, True)
+                self._models.set_item(k, v)
+                setattr(self, k, self.ModelStub(self, k))
+
+    def _set_nice_description(self, node, names):
+        ans = {'date': datetime.now().strftime('%Y-%m-%d %H:%M:%S'),
+               'dimensions': self._signal.axes_manager._get_dimension_str(),
+               }
+        node.add_dictionary(ans)
+        for n in names:
+            node.add_node('components.' + n)
+
+    def _save(self, name, dictionary):
+
+        from itertools import product
+        _abc = 'abcdefghijklmnopqrstuvwxyz'
+
+        def get_letter(models):
+            l = len(models)
```

Comment:
Bad convention to use `l` for variable name, since it is so similar to 1. Syntax highlighting fixes this, but it is not always available. Simple fix.